script/develop.sh

```diff
@@ -32,4 +32,4 @@ cleanup() {
 trap cleanup EXIT INT
 
 # start a watcher. This loops forever, so we don't need to loop ourselves.
-find src styleguide-app -type f -not -ipath '*elm-stuff*' | entr -c -p make public
+exec watchexec --clear --postpone -- shake --compact "$SHAKE_TARGET"
```

Comment:
`exec` here makes the trap above never hit. I was really confused why but [this SO](https://serverfault.com/questions/803247/parent-bash-script-not-receiving-trap-despite-process-still-running) made it clearer for me, *I think*:

> The trap command only works as long as the script is running.

I think the reason is `exec` makes `watchexec` the running process and `ctrl+c` is sent to it, instead of the script.

Was there a specific reason for adding `exec` here? I usually only do it if the launched process has some trouble without it (e.g. docker doesn't clean up on ctrl+c unless we `exec` it when doing `docker run`, I think).

I'm gonna go ahead and remove it, but feel free to add it back in case it's necessary!

Reply:
oh duh, sorry… I'm just used to adding `exec` in front of the last command if the point of the script is to set something up. I totally forgot about the teardown. Yes, you've got the reason right—`exec` makes the invoked command take over the whole process, so the bash `trap` never gets called.